Pages/ShoppingSite.ts

```diff
@@ -0,0 +1,94 @@
+import { Locator, Page } from '@playwright/test'
+const baseURL = 'https://react-shopping-cart-67954.firebaseapp.com/'
+export class ShoppingSite {
+  readonly page: Page
+  readonly addToCart: Locator
+  readonly closeCheckout: Locator
+  readonly cartAmount: Locator
+  readonly removeItemButton: Locator
+  readonly sizeLocator: (size: string) => Locator
+  readonly stockLevel: Locator
+  readonly openCheckout: Locator
+  readonly checkoutButton: Locator
+
+  constructor(page: Page) {
+    this.page = page
+    this.addToCart = page.getByRole('button', { name: 'Add to cart' })
+    this.closeCheckout = page.getByRole('button', { name: 'X' })
+    this.cartAmount = page.locator(`.sc-1h98xa9-9`)
+    this.openCheckout = page.locator('.sc-1h98xa9-2')
+    this.checkoutButton = page.getByRole('button', { name: 'Checkout' })
+    this.sizeLocator = (size: string) =>
+      page.getByText(`${size}`, { exact: true })
+    this.stockLevel = page.locator(`.sc-ebmerl-4 > p:nth-child(1)`)
+    this.removeItemButton = page.getByRole('button', {
+      name: 'remove product from cart',
+    })
+  }
+
+  async navigateToSite() {
+    await this.page.goto(baseURL)
+    await this.page.waitForTimeout(1000)
+  }
+
+  async selectSize(size: string) {
+    await this.page.waitForSelector(`text=${size}`, { state: 'visible' })
+
+    await this.page.waitForTimeout(500)
+
+    const sizeLocator = this.sizeLocator(size)
+
+    await sizeLocator.click()
+
+    await this.page.waitForTimeout(500)
+
+    await sizeLocator.click()
+  }
+
+  async getStockLevel() {
+    const stockText = (await this.stockLevel.textContent()) || '0'
+    await this.page.waitForLoadState('networkidle')
+    return parseInt(stockText, 10)
+  }
+
+  async clickCloseCheckout() {
+    await this.closeCheckout.click()
+  }
+
+  async getCartAmount() {
+    return await this.cartAmount.textContent()
+  }
+
+  async clickAddToCart(index: number) {
+    await this.addToCart.nth(index).click()
+  }
+
+  async countItemsInCart() {
+    return await this.removeItemButton.count()
+  }
+
+  async getItemsCount() {
+    return await this.addToCart.count()
+  }
+
+  async addMultipleItems(count: number) {
+    const items = await this.getItemsCount()
+    for (let i = 0; i < Math.min(count); i++) {
+      await this.clickAddToCart(i)
+      await this.clickCloseCheckout()
+    }
+  }
+
+  async clickOpenCheckout() {
+    await this.openCheckout.click()
+  }
+
+  async ClickRemoveItem(index: number) {
```

Comment:
camelCase :)

Reply:
I've started using that for naming folders too

Reply:
Fixed